jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/kubernetes/KubernetesUndeployService.java

```diff
@@ -108,7 +108,7 @@ private void undeployCustomResources(String namespace, File resourceDir, Resourc
   // Visible for testing
   String currentNamespace(List<HasMetadata> entities) {
     for (HasMetadata entity : entities) {
-      if (entity instanceof Namespace || entity instanceof Project) {
+      if ((entity instanceof Namespace || entity instanceof Project) && entity.getMetadata() != null) {
```

Comment:
When did you get a NullPointerException for this case?

Reply:
When I remove this I was getting NullPointerException due to this:
https://github.com/eclipse/jkube/blob/d2234119936e8ab95c258f4538f57deefb5fa075/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/KubernetesUndeployServiceTest.java#L93

Reply:
Ok, but that's because the Test is providing an invalid Project/Namespace.
In a real-world scenario, I understand that if this happens, we shouldn't ignore it. Unless I'm wrong, receiving a Namespace/Project lacking Metadata should be a flag of something going very wrong so maybe reporting as an Exception is something we want.
With the current approach, we're simply assuming there's no namespace.